### PR TITLE
Never Use Validation Set for Image Classification

### PR DESCRIPTION
**Pull request by TobyRoseman**
Fix for #519 and #419.

If the total number of images in the training set is at least 100, then by default logistic regression will create a validation set. Having a validation set causes second order statistics to be calculated at the end of training. This calculate is very costly when there are a large number of features. The fix is easy, just don't use a validation set.

### CODE REVIEW

**Comment by srikris** [comment]
I'm not sure this is the right fix. The second order information is being computed but never used. This is an O(N * F * F) operation where F is the number of features (which is around 1K).

```
// Compute validation-set stats.
if (lr_interface->num_validation_examples() > 0) {
         // Recycle lvalues from stats to use as out parameters here, now that we're
        // otherwise done reading from stats.
        lr_interface->compute_validation_second_order_statistics(
             stats.solution, stats.hessian, stats.gradient, stats.func_value);
             state["validation_loss"] =  stats.func_value;
}
```

This is an extremely expensive operation and I'm not sure why we are doing it at the end of training. The right fix should be to replace that function with

```
 lr_interface->compute_validation_first_order_statistics
```

If that function doesn't exist, we should add it since the code that follows doesn't actually need to compute the Hessian.

@znation This seems like a fix worth a point release.

**Comment by srikris** [comment]
Seems like it was introduced in https://github.com/apple/turicreate/commit/285cb3a1f65919b29afd5c34a8df0ea163677259

**Comment by srikris** [comment]
Please refer to #532 which keeps the validation tracking but fixes the underlying issue.
